crates/subspace-node/src/commands/run/consensus.rs

```diff
@@ -393,8 +393,8 @@ pub(super) struct ConsensusChainOptions {
     /// Create object mappings from the supplied piece index, or genesis if no piece index is
     /// specified. By default, mappings are disabled.
     ///
-    /// --dev mode enables mappings from genesis automatically.
-    #[arg(long, default_value_t, default_missing_value("0"))]
+    /// --dev mode enables mappings from genesis, unless a piece index has already been specified.
+    #[arg(long, default_value_t, default_missing_value("0"), num_args(0..=1))]
     create_object_mappings: CreateObjectMappings,
```

Comment:
I think what you want is to simply make it optional, then you can check for `None` and replace it with placeholder value when `--dev` is used or return an error to the user otherwise. See how it is done for `--chain` and `--reward-address` in the farmer.